metatensor-torch/src/block.cpp

```diff
@@ -331,6 +331,13 @@ TorchTensorBlock TensorBlockHolder::load_buffer(torch::Tensor buffer) {
 
 
 void TensorBlockHolder::save(const std::string& path) const {
+    // check that dtype is float64
+    if (this->scalar_type() != torch::kFloat64) {
+        C10_THROW_ERROR(ValueError,
+            "cannot save TensorBlock with dtype " + scalar_type_name(this->scalar_type()) +
+            ", only float64 is supported"
+        );
+    }
```

Comment:
should we also give a better error message for the device? I'm fairly sure the data needs to be on CPU.